homeassistant/components/aws/notify.py

```diff
@@ -0,0 +1,278 @@
+"""AWS platform for notify component."""
+import asyncio
+import logging
+import json
+import base64
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_PLATFORM, CONF_NAME, ATTR_CREDENTIALS
+from homeassistant.components.notify import (
+    ATTR_TARGET,
+    ATTR_TITLE,
+    ATTR_TITLE_DEFAULT,
+    BaseNotificationService,
+    PLATFORM_SCHEMA,
+)
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.json import JSONEncoder
+
+from .const import (
+    CONF_ACCESS_KEY_ID,
+    CONF_CREDENTIAL_NAME,
+    CONF_PROFILE_NAME,
+    CONF_REGION,
+    CONF_SECRET_ACCESS_KEY,
+    DATA_SESSIONS,
+)
+
+DEPENDENCIES = ["aws"]
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CONTEXT = "context"
+CONF_SERVICE = "service"
+
+SUPPORTED_SERVICES = ["lambda", "sns", "sqs"]
+
+
+def _in_avilable_region(config):
+    """Check if region is available."""
+    import aiobotocore
+
+    session = aiobotocore.get_session()
+    available_regions = session.get_available_regions(config[CONF_SERVICE])
+    if config[CONF_REGION] not in available_regions:
+        raise vol.Invalid(
+            "Region {} is not available for {} service, mustin {}".format(
+                config[CONF_REGION], config[CONF_SERVICE], available_regions
+            )
+        )
+    return config
+
+
+PLATFORM_SCHEMA = vol.Schema(
```

Comment:
We shouldn't allow this platform to be set up independently of the aws component. The notify schema is integrated in the component config schema, so we should not have a standalone platform schema here.

Reply:
I didn't mentioned that in the document, but my current implement indeed support both ways. Is that a no?

Reply:
Yes, we should consolidate config to one place.